### PR TITLE
[No ticket] add okta

### PR DESCRIPTION
**Pull request by MigGrover**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -2,12 +2,15 @@ package config
...
-	DB_name     string `mapstructure:"db_name"`
-	DB_ssl_mode string `mapstructure:"db_ssl_mode"`
-	Port        string `mapstructure:"port"`
-	Local       bool   `mapstructure:"local"`
+	Port         string `mapstructure:"port"`
+	Local        bool   `mapstructure:"local"`
+	ClientID     string `mapstructure:"client_id"`
+	ClientSecret string `mapstructure:"client_secret"`
+	Issuer       string `mapstructure:"issuer"`
+	DB_host      string `mapstructure:"db_host"`
```
Comment (alvarocueva) [inline config/config.go:10]: CamelCase, please

File: middleware/middleware.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	// Validate the header is present.
+	authHeader := c.GetHeader("Authorization")
+
+	if authHeader == "" {
+		om.logger.Error().Msg("Access token not found")
+		return false
+	}
+
+	email := c.Query("email")
```
Comment (alvarocueva) [inline middleware/middleware.go:36]: Add validation for email empty

File: middleware/middleware.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	if claims.Valid() != nil {
+		om.logger.Error().Msg("token expired")
+		return false
+	}
+
+	return true
+}
+
+func (om *Middleware) AuthMiddleware() gin.HandlerFunc {
```
Comment (alvarocueva) [inline middleware/middleware.go:68]: Add a description here

File: router/router.go [hunk 4]
```diff
@@ -10,14 +10,19 @@ type Controller interface {
...
 // New creates the router
-func New(controller Controller) *gin.Engine {
+func New(controller Controller, middleware Middleware) *gin.Engine {
 	r := gin.New()
 	root := r.Group("/")
 	root.GET("/healthz", controller.HealthCheck)
 
 	v1 := r.Group("/v1")
-	v1.GET("/users", controller.GetUsers)
+	v1.GET("/users", middleware.AuthMiddleware(), controller.GetUsers)
```
Comment (elCarlosHez) [inline router/router.go:25]: Nit comment, no blocker: We should use this `middleware` in the group instead of individual routes.

**Review by elCarlosHez — APPROVED** [review]

**Review by alvarocueva — APPROVED** [review]

**Pull request merged**